src/index.ts

```diff
@@ -96,6 +96,13 @@ export async function execute(filenames: string[], options: Partial<ExecutionOpt
     // execute them
     const interpreter = new Interpreter(executionOptions);
     interpreter.onError(logError);
+    // save each custom component def into a global map so we can access it
+    // at run time when we call `createObjectByType`
+    nodeDefs.map(node => {
+        if (node.isFulfilled) {
```

Comment:
I think you also need to check if the promise wasn't rejected or you may get an `undefined is not an object` error